net/FlatBuffers/Table.cs

```diff
@@ -83,15 +84,15 @@ protected int __vector(int offset)
         }
 
         // Initialize any Table-derived type to point to the union at the given offset.
-        protected TTable __union<TTable>(TTable t, int offset) where TTable : Table
+        public T __union<T>(int offset) where T : struct, Accessor
```

Comment:
Issue with struct implementing an interface is that we may start running into boxing (and thus performance) issues.

https://blogs.msdn.microsoft.com/abhinaba/2005/10/05/c-structs-and-interface/